src/containers/Task/Task.jsx

```diff
@@ -2,139 +2,95 @@ import React, { useState, useEffect } from 'react';
+  const isTaskSubmitted = task   => tasksSubmitted.includes(task?.identifier);
+  const extractTask     = task   => !task || isTaskSubmitted(task)? setTaskUrl('') : setTask(task);
+  
+  const getCampaignCallback = ({ ControlAction: campaign }) => setCampaign(campaign[0]);
+  const getTaskCallback     = ({ ControlAction: task })     => extractTask(task[0]);
+  const getTaskListCallback = ({ ControlAction: taskList }) => {
+    if (taskList.length){
+      const selection = randomSelection(taskList).map(({ identifier }) => identifier);
+
+      setTasksTodo(selection);
+      setTaskUrl(selection[0]);
     } else {
-      history.replace(`/campaign/${campaignIdentifier}/task/`);
+      setTask(null);
+      setTasksTodo([]);
+      setTimeout(() => setRefetchCount(refetchCount + 1), refetchTimeout);
     }
   };
+  
+  const [getCampaign, { loading:loadingCampaign }] = useLazyQuery(GET_CAMPAIGN, { onCompleted: getCampaignCallback });
+  const [getTask,     { loading:loadingTask }]     = useLazyQuery(NEXT_TASK,    { onCompleted: getTaskCallback });
+  const [getTaskList, { loading:loadingTaskList }] = useLazyQuery(ALL_TASKS,    { onCompleted: getTaskListCallback, fetchPolicy: "no-cache" });
+  
+  const onGoBackClick             = () => history.goBack();
+  const handleNextTaskButtonClick = () => {
+    addToSubmitted(taskIdentifier);
+    removeFromTodo(taskIdentifier);
+    setTaskUrl(getNextTask(taskIdentifier));
+  };
 
-  // Get campaign first
   useEffect(() => {
     if (campaignIdentifier) {
-      client.query({
-        query    : GET_CAMPAIGN,
-        variables: { identifier: campaignIdentifier },
-      })
-        .then(response => {
-          setCampaign(response.data.ControlAction[0]);
-        });
+      getCampaign({ variables: { identifier: campaignIdentifier } });
     }
-  }, [campaignIdentifier]);
+  }, [campaignIdentifier, getCampaign]);
 
-  // Retrieve task on existing draw (task identifier in url)
   useEffect(() => {
-    if (taskIdentifier) {
-      client.query({
-        query    : CLIENT_NEXT_POTENTIAL_ACTION_QUERY,
-        variables: { taskIdentifiers: [taskIdentifier], offset: 0 },
-      })
-        .then(response => {
-          const newTaskNotReceived = response.data.ControlAction.find(({ identifier }) => !tasksReceived.includes(identifier));
-
-          if (taskTotal === tasksReceived.length) {
-            return;
-          } else if (newTaskNotReceived) {
-            setTask(newTaskNotReceived);
-          } else {
-            history.replace(`/campaign/${campaignIdentifier}/task`);
-          }
-        });
-    }
-  }, [campaignIdentifier, history, taskIdentifier, taskTotal, tasksReceived]);
-
-  // Retrieve tasks on new draw (task identifier not in url)
-  useEffect(() => {
-    if (taskIdentifier || loading || !campaign) {
-      return;
+    if (!taskIdentifier && score) {
+      getTaskList({ variables: { identifier: score.identifier, filtered: tasksSubmitted, drawamount: taskDrawAmount } });
```

Comment:
Is it required to make drawamount camelCasing here?

Reply:
Good point. It's not required, but it's allowed, so it makes sense to keep consistency